eventrecorder/recorder.go

```diff
@@ -66,7 +69,8 @@ func (r *EventRecorder) httpServerMux() *http.ServeMux {
 	mux := http.NewServeMux()
 	mux.HandleFunc("/v1/retrieval-events", r.handleRetrievalEvents)
 	mux.HandleFunc("/ready", r.handleReady)
-	mux.HandleFunc("/v1/summarize-and-clear", r.handleStats)
+	// register prometheus metrics
+	mux.Handle("/metrics", metrics.NewExporter())
```

Comment:
This exposes metics publicly. Instead, metrics should be exposed on a separate HTTP server accessible only internally, which then gets scraped by a `ServiceMonitor` or `PodMonitor` CRD.